From a79e900f289c1a6e3dcfd7938b46b2c315100770 Mon Sep 17 00:00:00 2001
From: John Zhuge <jzhuge@cloudera.com>
Date: Wed, 17 May 2017 20:12:53 -0700
Subject: [PATCH 2444/2848] HADOOP-14197. Fix ADLS doc for credential
 provider. Contributed by John Zhuge

(cherry picked from commit c55195588f5853b16d34cb6389cc6f84acc3edea)

Change-Id: Ib481b2b80ab2ce7dce2cdcd9de3758eda35f0500
---
 .../src/site/markdown/index.md                     |   15 ++++++++++-----
 1 file changed, 10 insertions(+), 5 deletions(-)

diff --git a/hadoop-tools/hadoop-azure-datalake/src/site/markdown/index.md b/hadoop-tools/hadoop-azure-datalake/src/site/markdown/index.md
index 3c4c70e..de8fda2 100644
--- a/hadoop-tools/hadoop-azure-datalake/src/site/markdown/index.md
+++ b/hadoop-tools/hadoop-azure-datalake/src/site/markdown/index.md
@@ -166,9 +166,9 @@ For additional reading on the credential provider API, see
 ##### Provisioning
 
 ```bash
-hadoop credential create dfs.adls.oauth2.refresh.token -value 123
+hadoop credential create dfs.adls.oauth2.client.id -value 123
     -provider localjceks://file/home/foo/adls.jceks
-hadoop credential create dfs.adls.oauth2.credential -value 123
+hadoop credential create dfs.adls.oauth2.refresh.token -value 123
     -provider localjceks://file/home/foo/adls.jceks
 ```
 
@@ -176,6 +176,10 @@ hadoop credential create dfs.adls.oauth2.credential -value 123
 
 ```xml
 <property>
+  <name>dfs.adls.oauth2.access.token.provider.type</name>
+  <value>RefreshToken</value>
+</property>
+<property>
   <name>hadoop.security.credential.provider.path</name>
   <value>localjceks://file/home/foo/adls.jceks</value>
   <description>Path to interrogate for protected credentials.</description>
@@ -186,9 +190,10 @@ hadoop credential create dfs.adls.oauth2.credential -value 123
 
 ```bash
 hadoop distcp
-    [-D hadoop.security.credential.provider.path=localjceks://file/home/user/adls.jceks]
-    hdfs://<NameNode Hostname>:9001/user/foo/007020615
-    adl://<Account Name>.azuredatalakestore.net/testDir/
+    [-D dfs.adls.oauth2.access.token.provider.type=RefreshToken
+     -D hadoop.security.credential.provider.path=localjceks://file/home/user/adls.jceks]
+    hdfs://<NameNode Hostname>:9001/user/foo/srcDir
+    adl://<Account Name>.azuredatalakestore.net/tgtDir/
 ```
 
 NOTE: You may optionally add the provider path property to the `distcp` command
-- 
1.7.9.5

